### PR TITLE
WPC-657: As a publisher, when uploading an image, i want the metadata automatically captured

### PR DESCRIPTION
**Pull request by elias-summermatter**

### CODE REVIEW

File: packages/editor/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "watch:client": "webpack-dev-server --mode development --hot --inline",
     "watch:server": "run-p watch:server:*",
     "watch:server:start": "nodemon -r dotenv/config --inspect=0.0.0.0:9230 ./dist/server/index.js",
-    "watch:server:build": "tsc -b ./src/shared ./src/server --watch",
+    "watch:server:build": "tsc -b ./src/shared ./src/server ./src/client --watch",
```
Comment (michael-scheurer) [inline packages/editor/package.json:40]: why?

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx [hunk 2]
```diff
@@ -2,21 +2,90 @@ import React, {useState} from 'react'
...
 import {ImageUploadPanel} from './imageUploadPanel'
+import exifr from 'exifr'
 
 export interface ImageUploadAndEditPanelProps {
   onClose(): void
   onUpload(): void
 }
 
+export type ImageMetaData = {
+  [key: string]: string
```
Comment (michael-scheurer) [inline packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:13]: Can't we use the meta data type from the exifr library?

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx [hunk 3]
```diff
@@ -2,21 +2,90 @@ import React, {useState} from 'react'
...
+}
+
+type metaTagsMap = {
+  [key: string]: string[]
+}
+
+const DEFAULT_META_TAG_MAP: metaTagsMap = {
```
Comment (michael-scheurer) [inline packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:20]: idem

File: packages/editor/package.json (hunk 1 above)
Comment (elias-summermatter) [reply packages/editor/package.json:40]: For me the hot reload of the editor did not work without this.... Does anyone has the same problem?

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx (hunk 2 above)
Comment (elias-summermatter) [reply packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:13]: What do you mean? Its nothing that existed... Its something that i coded and has nothing to do with an existing library... Or did i get you wrong?

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx (hunk 3 above)
Comment (elias-summermatter) [reply packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:20]: idem

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx (hunk 2 above)
Comment (elias-summermatter) [reply packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:13]: @michael-scheurer as discusted over the phone :-)

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx (hunk 3 above)
Comment (elias-summermatter) [reply packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:20]: @michael-scheurer as discusted over the phone :-)

**Review by michael-scheurer — APPROVED** [review]

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx [hunk 4]
```diff
@@ -2,21 +2,90 @@ import React, {useState} from 'react'
...
+      if (base[nestedTag]) {
+        base = base[nestedTag]
+      } else {
+        return false
+      }
+    }
+    return base
+  }
+
+  async function readPictureMetaData(data: File): Promise<ImageMetaData> {
```
Comment (tomaszdurka) [inline packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:55]: for consistency rename to `readImageMetaData`

File: packages/editor/package.json (hunk 1 above)
Comment (tomaszdurka) [inline packages/editor/package.json:40]: If you want to have this add another command for consistency.
`"watch:client:build": "tsc -b ./src/client --watch"` plus `watch:client:start` and rework `watch:client`...

**Review by tomaszdurka — CHANGES_REQUESTED** [review]

**Review by camilleory — APPROVED** [review]

**Review by tomaszdurka — DISMISSED** [review]

**Review by michael-scheurer — APPROVED** [review]

**Review by camilleory — APPROVED** [review]

**Review by tomaszdurka — APPROVED** [review]

**Comment by elias-summermatter** [comment]
@tomaszdurka can you please approve or comment

**Comment by elias-summermatter** [comment]
@michael-scheurer can you have a look and if fine aprove

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx [hunk 5]
```diff
@@ -2,21 +2,90 @@ import React, {useState} from 'react'
...
+    source: '',
+    link: '',
+    licence: ''
+  })
+  async function handleUpload(file: File) {
+    setImageMetaData(await readImageMetaData(file))
     setFile(file)
   }
 
+  function findNestedMetaFields(tags: any, tag: string) {
```
Comment (michael-scheurer) [inline packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:42]: Pl. provide minimal description.

File: packages/editor/src/client/panel/imageUploadAndEditPanel.tsx [hunk 6]
```diff
@@ -2,21 +2,90 @@ import React, {useState} from 'react'
...
+      if (base[nestedTag]) {
+        base = base[nestedTag]
+      } else {
+        return false
+      }
+    }
+    return base
+  }
+
+  async function readImageMetaData(data: File): Promise<ImageMetaData> {
```
Comment (michael-scheurer) [inline packages/editor/src/client/panel/imageUploadAndEditPanel.tsx:55]: Pl. provide minimal description.

**Review by michael-scheurer — APPROVED** [review]

**Review by camilleory — APPROVED** [review]

**Review by tomaszdurka — CHANGES_REQUESTED** [review]
@elias-summermatter we have alert here, can you adjust your code to remove that one?

https://github.com/wepublish/wepublish/pull/545/checks?check_run_id=6005499329

**Review by tomaszdurka — APPROVED** [review]

**Pull request merged**